app/src/main/java/com/douglasbruce/tasky/core/network/interceptor/JwtInterceptor.kt

```diff
@@ -1,12 +1,18 @@
 package com.douglasbruce.tasky.core.network.interceptor
 
+import androidx.datastore.core.DataStore
+import com.douglasbruce.tasky.UserPreferences
+import kotlinx.coroutines.flow.map
 import okhttp3.Interceptor
 import okhttp3.Response
+import javax.inject.Inject
 
-class JwtInterceptor : Interceptor {
+class JwtInterceptor @Inject constructor(
+    private val userPreferences: DataStore<UserPreferences>,
+) : Interceptor {
 
     override fun intercept(chain: Interceptor.Chain): Response {
-        val token: String? = null
+        val token = userPreferences.data.map { it.token }
```

Comment:
`userPreferences.data` is a `Flow` right? So this wouldn't be the token, but a `Flow<String>` if I'm not wrong. To get the token, you'd need something like this:
```
val token = runBlocking {
    userPreferences.data.map { it.token }.first()
}
```